vulnerabilities/importers/apache_httpd.py

```diff
@@ -81,56 +87,82 @@ def to_advisory(self, data):
-                    PackageURL(type="apache", name="httpd", version=version)
-                    for version in self.version_api.get("apache/httpd").valid_versions
-                    if SemverVersion(version) in version_range
-                ]
+        affected_packages = []
+        if affected_version_range:
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(
+                        type="generic",
+                        name="apache_httpd",
+                    ),
+                    affected_version_range=affected_version_range,
+                )
             )
 
         return AdvisoryData(
-            vulnerability_id=cve,
+            aliases=[alias],
             summary=description,
-            affected_packages=nearest_patched_package(affected_packages, fixed_packages),
+            affected_packages=affected_packages,
             references=[reference],
         )
 
-    def to_version_ranges(self, versions_data):
-        fixed_version_ranges = []
-        affected_version_ranges = []
+    def to_version_ranges(self, versions_data, fixed_versions):
+        constraints = []
         for version_data in versions_data:
             version_value = version_data["version_value"]
             range_expression = version_data["version_affected"]
-            if range_expression == "<":
-                fixed_version_ranges.append(
-                    VersionRange.from_scheme_version_spec_string(
-                        "semver", ">={}".format(version_value)
+            if range_expression not in KNOWN_RANGE_EXPRESSIONS:
+                raise ValueError(f"unknown comparator found! {range_expression}")
+            if range_expression == ">=" or range_expression == "!<":
+                constraints.append(
+                    VersionConstraint(
+                        comparator=">=",
+                        version=SemverVersion(version_value),
+                    )
+                )
+
+            if range_expression == "<=":
+                constraints.append(
+                    VersionConstraint(
+                        comparator="<=",
+                        version=SemverVersion(version_value),
                     )
                 )
-            elif range_expression == "=" or range_expression == "?=":
-                affected_version_ranges.append(
-                    VersionRange.from_scheme_version_spec_string(
-                        "semver", "{}".format(version_value)
+
+            if range_expression == "?=":
+                pass
+
+            if range_expression == "=":
+                constraints.append(
+                    VersionConstraint(
+                        comparator="=",
+                        version=SemverVersion(version_value),
                     )
                 )
 
-        return (fixed_version_ranges, affected_version_ranges)
+        for fixed_version in fixed_versions:
+            constraints.append(
```

Comment:
Invert() is not obvious: a comment explaining what you are doing would be useful for the reader

Reply:
Done.  (My comment might be a bit verbose -- improvements are welcome!)